openquakeplatform/openquakeplatform/static/js/risk_input_model_fragility.js

```diff
@@ -0,0 +1,300 @@
+
+    // Create the table containers, as many as the user wants
+    count += 1;
+
+    var imls;
+    // Imls value needs to be an array for discrete functions,
+    // and minIML & maxIML for continuous
+    if (fFormat == 'discrete') {
+        imls =
+            '<label>IML: </label>' +
+            '<input id="'+count+'" class="imls ffsTable" placeholder="imls array" type="text">';
+    } else if (fFormat == 'continuous') {
+        imls =
+            '<label> MinIML: </label>' +
+            '<input id="'+count+'" class="minImls ffsTable" type="text"><br>' +
+            '<label> MaxIML: </label>' +
+            '<input id="'+count+'" class="maxImls ffsTable" type="text">';
+    }
+
+    // Create the fragility function set (ffs)
+    $('#tables').append(
+        '<div id="table'+count+'" class="ffsTableDiv panel panel-default">' +
+        '<strong class="ffsTitle">'+fFormat.toUpperCase()+'</strong><button id="'+count+'" class="btn-danger btn destroyTable">Remove</button><br>' +
+            '<div class="ffsForm" >' +
+                '<label> Function Id: </label>' +
+                '<input id="'+count+'" class="ffsIds ffsTable" type="text"><br>' +
+                '<label> IMT: </label>' +
+                '<input id="'+count+'" class="imt ffsTable" type="text" placeholder="PGA"><br>' +
+                '<label> Damage Limit: </label>' +
+                '<input id="'+count+'" class="noDamageLimit ffsTable" type="text"><br>' +
+                imls +
+                '<input id="'+count+'" type="hidden" class="fFormat ffsTable" value="'+fFormat+'" >' +
+                '<br>' +
+            '</div>'+
+            '<div id="tableDiv'+count+'" class="theTable"></div><br><br>' +
+            '<br><br>' +
+        '</div>'
+    );
+
+    // force bootstrap style
+    $('.btn-danger').css({'background-color': '#da4f49'});
+
+    var container = document.getElementById('tableDiv'+count);
+
+
+    //////////////////////
+    /// Table Settings ///
+    //////////////////////
+
+    table = new Handsontable(container, {
+        colHeaders: header,
+        startCols: headerLength,
+        maxCols: headerLength,
+        startRows: limitStateLength,
+        colWidths: colWidth,
+    });
+
+    // Populate the table with limit states
+    for (var i = 0; i < limitStates.length; i++) {
+        table.setDataAtCell(i, 0, limitStates[i]);
+    }
+
+    activeTablesObj[count] = table;
+
+    $('#outPutFF').empty();
+    $('#saveBtnFF').css('display', 'block');
+
+    // Logic to remove a table
+    $('.destroyTable').click(function() {
+        $('#table'+this.id).remove();
+        var removedTable = this.id;
+        delete activeTablesObj[removedTable];
+    });
+
+    // Increase the ffs panel when many limit states are defined
+    if (limitStateLength > 5) {
+        console.log('limitStateLength:');
+        console.log(limitStateLength);
+        $('.ffsTableDiv').height(240 + (limitStateLength * limitStateLength));
```

Comment:
I am not sure I am reading this correctly, but it looks like you should set both the height and width, instead of setting the height to the square of limitStateLength.